CMakeLists.txt

```diff
@@ -38,51 +39,46 @@ if(${enable_cbmc_tests})
     enable_testing()
 endif()
 
-file(STRINGS src/config.inc config_inc_v REGEX "CBMC_VERSION *= *[0-9\.]+")
-string(REGEX REPLACE "^CBMC_VERSION *= *" "" CBMC_RELEASE ${config_inc_v})
-message(STATUS "CBMC release ${CBMC_RELEASE}")
-
+# based on https://cmake.org/pipermail/cmake/2010-July/038015.html
 find_package(Git)
+if(GIT_FOUND)
+  file(WRITE ${CMAKE_BINARY_DIR}/version.cmake
+    "
+    file(STRINGS ${CMAKE_SOURCE_DIR}/src/config.inc
```

Comment:
If I understand the cmake manual correctly, this needs to be `CMAKE_CURRENT_SOURCE_DIR`, otherwise this will be relative to the project root (which need not be the root of this repo).

Reply:
If you include the cbmc repository as a submodule in another repository (typically in a subdirectory, e.g. `lib/cbmc`) then this will not work because it will look for `config.inc` in the wrong place.

Reply:
I'm sorry, I hadn't thought about this, it's of course an essential use case for you. I have now implemented a different solution, which uses `CBMC_SOURCE_DIR`. I believe this should actually look in the correct folder whatever the project.